packages/api-explorer/src/JsonForm/anyOf-custom-editor.js

```diff
@@ -0,0 +1,168 @@
+      classNames: {
+        listItems: 'multiple-select-choices-list choices__list--multiple',
+        item: 'multiple-select-item choices__item'
+      }
+    })
+
+    switcher.addEventListener('change', e => {
+      e.preventDefault();
+      e.stopPropagation();
+      const selectedValues = Array.from(e.currentTarget.selectedOptions).map(selectedValue => selectedValue.value)
+      if (selectedValues.length === 0) {
+        self.showErrorMessage('empty-selection')
+      } else if (selectedValues.length === 1) {
+        self.hideErrorMessage()
+        self.switchEditor(self.display_text.indexOf(selectedValues[0]))
+      } else {
+        self.updateEditor(selectedValues)
+      }
+      self.onChange(true)
+      choices.hideDropdown()
+    });
+
+    return { switcher, switcherDiv }
+  },
+  updateEditor(selectedValues) {
+    const joinedValues = selectedValues.join('_')
+    const mergedEditorIndex = editorIndexesMap.get(joinedValues)
+    if (mergedEditorIndex) {
+      this.hideErrorMessage()
+      this.switchEditor(mergedEditorIndex);
+      return
+    }
+    const schemasIndexes = selectedValues.map(value => this.display_text.indexOf(value))
+    const schemas = this.types.filter((_, index) => schemasIndexes.includes(index))
+    const mergedSchemas = mergeSchemas(schemas, self)
+    if (!mergedSchemas) {
+      this.showErrorMessage('not-compatible')
+      return
+    }
+    this.hideErrorMessage()
+    const index = this.types.length
+    this.insertNewEditor(index, joinedValues, mergedSchemas)
+    this.switchEditor(index)
+  },
+  addErrorMessageHtmlNode() {
+    this.errorMessageHtmlNode = document.createElement('p')
+    this.errorMessageHtmlNode.style.color = 'red'
+    this.errorMessageHtmlNode.style.fontWeight = 'bold'
+    this.errorMessageHtmlNode.style.display = 'none'
+    this.errorMessageHtmlNode.style.marginTop = '10px'
+    this.container.appendChild(this.errorMessageHtmlNode)
+  },
+  showErrorMessage(type) {
+    const  errorsByType = {
+      'not-compatible': 'The selected schemas are not compatible!',
+      'empty-selection': 'Select at least 1 schema',
+      'start-config': ' '
+    }
+    this.errorMessageHtmlNode.innerText = errorsByType[type] || 'Unknown error!'
+    this.editor_holder.style.display = 'none'
+    this.errorMessageHtmlNode.style.display = 'block'
+  },
+  hideErrorMessage() {
+    this.editor_holder.style.display = 'block'
+    this.errorMessageHtmlNode.style.display = 'none'
+  },
+  insertNewEditor(index, joinedValues, mergedSchemas) {
+    editorIndexesMap.set(joinedValues, index)
+    this.types.push(mergedSchemas)
+    this.editors.push(false)
+  }
+})
+
+const editorIndexesMap = new Map()
+
+const mergeSchemas = (schemas) => {
+  const schemasType = schemas.reduce((type, currentSchema) => {
+    if (type === 'any') {
+      return currentSchema.type
+    }
+    return currentSchema.type === type
+      ? type
+      : undefined
+  }, 'any')
```

Comment:
This function is quite cryptical, could you explain what should do?

Reply:
I refectored it a bit. Hopefully it should be clearer now. It merges the input openapi schemas

Reply:
I'm checking on the refactor but as far as I can see you simply moved this reduce in a specific function, the behaviour is still pretty cryptical; what is this reduced used for? Getting the last "valid" value of the schema.type?

Reply:
I've asked @silversoul93 to review the newer implementation.